ADR/ADR038-add-psql-database-and-solid-queue-to-forms-runner.md

```diff
@@ -0,0 +1,90 @@
+
+Submission data would be kept in the database until we are sure the email has not bounced. We should be able to receive notifications about successful deliveries to an SNS topic, although further investigation is required into the reliability of this mechanism. We can either delete the data when we receive notification of a successful delivery, or when sufficient time has passed and we have not received a bounce/complaint notification. Forms Runner will need a background job to poll SQS queues subscribed to these SNS topics if we want to perform actions for these events. It makes sense to use ActiveJob to run a recurring task to do this.
+
+Adding a database now will allow for other features down the line. In particular, storing submission data in a database will likely be required when we implement the ability for form fillers to save their data and return later.
+
+It makes most sense for us to store submission data in a PostgreSQL database, as this provides reliable storage that is easily accessible and we already use PostgreSQL for other microservices.
+
+We have some concerns about database performance if we use the same Amazon Aurora instance for databases for all of our microservices. One app seeing increased request volumes places increased load on the database, which can degrade the performance of the other apps as the database slows. We also have some concerns about database restoration/roll back if all apps use the same Aurora instance. If this were the case we can only rollback databases for all our apps. The risk of losing submission data if this were to happen is a major concern. For these reasons we want to add a separate Aurora cluster for the Forms Runner database.
+
+We should consider encrypting the submission data at rest in the database, if we think that this adds worthwhile protection.
+
+## Decision
+
+We will use Solid Queue and ActiveJob to manage the background processing of emails.
+
+We will use PostgreSQL/RDS for the database on the Forms Runner, with a separate Aurora instance.
+
+## Consequences
+
+We will have a reliable way to deliver emails to form processors with files attached to the emails.
+
+Forms Runner will be responsible for running background jobs to send submission emails and consume SQS messages that inform us about the send result. We will have to manage any performance impacts of this and configure additional monitoring.
+
+We have an additional infrastructure dependency on PostgreSQL for Forms Runner. There will be additional costs to configuring another Amazon Aurora instance.
```

Comment:
Something in the back of our (infra) minds is RDS blue/green deployments. We want to avoid downtime with our current database, but we haven't had the capacity to work on this.
Having a database that submissions rely on probably increases the need for blue/green deployments

Reply:
Thanks, I can add that to the consequences

Reply:
Have added something about this.